internals/overlord/servstate/manager_test.go

```diff
@@ -1776,6 +1784,79 @@ services:
 	})
 }
 
+func (s *S) TestWorkloadAppliesToService(c *C) {
+	plan.RegisterSectionExtension(workloads.WorkloadsField, &workloads.PlanExtension{})
+	defer plan.UnregisterSectionExtension(workloads.WorkloadsField)
+
+	s.newServiceManager(c)
+	s.planAddLayer(c, `
+services:
+    test1:
+        override: replace
+        command: /bin/sh -c "echo $PATH; sleep 10"
+        workload: wl1
+
+workloads:
+    wl1:
+        override: replace
+        environment:
+            PATH: "/private/bin:/bin:/sbin"
+    `)
+	s.planChanged(c)
+
+	chg := s.startServices(c, [][]string{{"test1"}})
+	s.waitUntilService(c, "test1", func(svc *servstate.ServiceInfo) bool {
+		return svc.Current == servstate.StatusActive
+	})
+	c.Assert(s.manager.BackoffNum("test1"), Equals, 0)
+	s.st.Lock()
+	c.Check(chg.Status(), Equals, state.DoneStatus)
+	s.st.Unlock()
+	time.Sleep(10 * time.Millisecond)
+	c.Check(s.readAndClearLogBuffer(), Matches, `(?s).* \[test1\] /private/bin:/bin:/sbin\n`)
+}
+
+func (s *S) TestWorkloadReferenceInvalid(c *C) {
+	plan.RegisterSectionExtension(workloads.WorkloadsField, &workloads.PlanExtension{})
+	defer plan.UnregisterSectionExtension(workloads.WorkloadsField)
+
+	s.newServiceManager(c)
+	err := s.tryPlanAddLayer(c, `
+services:
+    test1:
+        override: replace
+        command: /bin/sh -c "echo $PATH; sleep 10"
+        workload: non-existing
+    `)
+	c.Assert(err, ErrorMatches, `plan service "test1" workload not defined: "non-existing"`)
+}
+
+func (s *S) tryPlanAddLayer(c *C, layerYAML string) error {
+	cnt := len(s.plan.Layers)
+	layer, err := plan.ParseLayer(cnt, fmt.Sprintf("test-plan-layer-%v", cnt), []byte(layerYAML))
+	if err != nil {
+		return err
+	}
+	// Resolve {{.NotifyDoneCheck}}
```

Comment:
What's this? Is this just a comment or does it do anything?

Reply:
It's just a comment, this was copy-pasted from `func (s *S) planAddLayer()`. `tryPlanAddLayer()` does almost the same, except it will return errors instead of asserting they are `nil`